src/main/scala/beam/agentsim/scheduler/BeamAgentScheduler.scala

```diff
@@ -181,10 +169,12 @@ class BeamAgentScheduler(val beamConfig: BeamConfig, stopTick: Double, val maxWi
       }
       val completionTickOpt = triggerIdToTick.get(triggerId)
       if (completionTickOpt.isEmpty || !triggerIdToTick.contains(triggerId) || !awaitingResponse
-            .containsKey(completionTickOpt.get)) {
+        .containsKey(completionTickOpt.get)) {
```

Comment:
A bit of an aside for this one, but seems like it wouldn't hurt here. Why not use pattern matching here...as well as the `contains` seems duplicative.

Reply:
Yes, pattern matching would be better here, but I don't want to touch anything extra because there are no tests for `BeamAgentScheduler`

Reply:
Hm, I was wrong, the tests are in `BeamAgentSchedulerSpec`